src/main/java/org/jabref/gui/preftabs/TableColumnsTab.java

```diff
@@ -438,14 +389,51 @@ public void setValues() {
         List<String> lengths = prefs.getStringList(JabRefPreferences.COLUMN_WIDTHS);
         for (int i = 0; i < names.size(); i++) {
             if (i < lengths.size()) {
-                tableRows.add(new TableRow(names.get(i), Integer.parseInt(lengths.get(i))));
+                tableRows.add(new TableRow(names.get(i), Double.parseDouble(lengths.get(i))));
             } else {
                 tableRows.add(new TableRow(names.get(i)));
             }
         }
         rowCount = tableRows.size() + 5;
-        ncWidth = prefs.getInt(JabRefPreferences.NUMBER_COL_WIDTH);
+    }
+
+    /*** end: special fields ***/
+
+    static class TableRow {
```

Comment:
Any reason why this has to be static?

Reply:
No idea... (the code quality in the preferences tabs is so bad anyway).

Reply:
I personally see to reason to have this whole column stuff in the preferences anylonger. I think it was only added because we could not save the column reordering the user did directly on the maintable. I would vote to remove that stuff completely.

Reply:
Well, currently, this is the only place where you can choose which columns are shown.

Reply:
Why not adding a table column selector like it's possible in nearly every table 

![grafik](https://user-images.githubusercontent.com/320228/36739548-8079be70-1be0-11e8-9d63-03d3c064e484.png)

Reply:
Here is an example https://gist.github.com/Roland09/d92829cdf5e5fee6fee9

Reply:
This is a good idea! I have to think a bit more about it and how to make it work for JabRef (we allow also columns of the form `author/editor` and have to many fields to display them all in a drop down - possible a combination of drop down and text field may work). But not as part of this PR.